libflux/src/core/ast/mod.rs

```diff
@@ -555,18 +556,31 @@ pub struct FunctionType {
 }
 
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[serde(tag = "type")]
 pub enum ParameterType {
+    #[serde(rename = "Required")]
     Required {
+        #[serde(skip_serializing_if = "BaseNode::is_empty")]
+        #[serde(default)]
+        #[serde(flatten)]
         base: BaseNode,
         name: Identifier,
         ty: MonoType,
     },
+    #[serde(rename = "Optional")]
     Optional {
+        #[serde(skip_serializing_if = "BaseNode::is_empty")]
+        #[serde(default)]
+        #[serde(flatten)]
         base: BaseNode,
         name: Identifier,
         ty: MonoType,
     },
+    #[serde(rename = "Pipe")]
     Pipe {
+        #[serde(skip_serializing_if = "BaseNode::is_empty")]
+        #[serde(default)]
+        #[serde(flatten)]
         base: BaseNode,
         name: Option<Identifier>,
```

Comment:
Could you add `#[serde(skip_serializing_if = "Option::is_none")]` right above this line so that we don't include a `name` field in the json if this is an un-named pipe parameter?